notification/notification.js

```diff
@@ -49,7 +49,22 @@ function handleMessage(request) {
     // TODO function to update results page
     showNotification(currentUrl, betterAlternative);
 }
-browser.runtime.onConnect.addListener( m => m.onMessage.addListener(handleMessage));
+
+function onError(){
+    showNotification(null);
+}
+
+// if the notifications are not paused 
+ if ( notification === "on")   { 
+    
+    browser.runtime.onConnect.addListener( m => m.onMessage.addListener(handleMessage)); 
```

Comment:
This line adds the listener to show the notifications, and this is only read and executed once when the extension is loaded. So we always need to register the listener or won't have the notifications functionality at all.
The notification variable is not declared anywhere in this file?

So I would move the check inside the showNotification or handleMessage functions.

```
    //if notifications are paused don't show: return
    if (localStorage.getItem("notification") === "off") return;
```